### PR TITLE
Shader manager cleanup

### PR DESCRIPTION
**Pull request by hrydgard**
This is really just some preparation for shader ID disk cache (store used shader IDs to disk, precompile  shaders used on the last run on startup to avoid stalls) which I have in another branch.

### CODE REVIEW

File: GPU/GLES/ShaderManager.cpp [hunk 1]
```diff
@@ -870,28 +870,24 @@ LinkedShader *ShaderManager::ApplyFragmentShader(Shader *vs, int prim, u32 vertT
 
 	if (ls == nullptr) {
 		// Check if we can link these.
-#ifdef _DEBUG
-		if (!DebugAreShadersCompatibleForLinking(vs, fs)) {
-			return nullptr;
-		}
-#endif
-		ls = new LinkedShader(vs, fs, vertType, vs->UseHWTransform(), lastShader_);  // This does "use" automatically
+		ls = new LinkedShader(VSID, vs, FSID, fs, vs->UseHWTransform());  // This does "use" automatically
```
Comment (unknownbrackets) [inline GPU/GLES/ShaderManager.cpp]: The comment is a lie.

-[Unknown]

File: lang [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 68c2f7139d64b99d3fa84769ea53b5a388a7f90e
+Subproject commit 84fd93497b9718a89ac04edfe9636b0f90943505
```
Comment (unknownbrackets) [inline lang]: Probably a mistake?  Seems this makes it older.

-[Unknown]

File: GPU/Common/ShaderId.cpp [hunk 3]
```diff
@@ -102,17 +102,18 @@ void ComputeVertexShaderID(ShaderID *id_out, u32 vertType, bool useHWTransform)
...
 			// doShadeMapping is stored as UVGenMode, so this is enough for isLightingEnabled.
-			if (gstate.isLightingEnabled())
+			if (gstate.isLightingEnabled()) {
+				id.SetBits(VS_BIT_MATERIAL_UPDATE, 3, gstate.getMaterialUpdate() & 7);
 				id.SetBit(VS_BIT_LIGHTING_ENABLE);
+			}
 			// Light bits
 			for (int i = 0; i < 4; i++) {
-				id.SetBit(VS_BIT_LIGHT0_ENABLE + i, gstate.isLightChanEnabled(i) != 0);
+				id.SetBit(VS_BIT_LIGHT0_ENABLE + i, gstate.isLightChanEnabled(i) != 0 && gstate.isLightingEnabled());
```
Comment (unknownbrackets) [inline GPU/Common/ShaderId.cpp]: Hmm.  I think we want:

```c++
bool chanEnabled = gstate.isLightChanEnabled(i) != 0 && gstate.isLightingEnabled();
id.SetBit(VS_BIT_LIGHT0_ENABLE + i, chanEnabled);
if (chanEnabled || (doShadeMapping && (gstate.getUVLS0() == i || gstate.getUVLS1() == i))) {
```

Not that I know if it ever happens, but it probably does...

-[Unknown]

Comment (hrydgard) [inline GPU/Common/ShaderId.cpp]: Hm, yes, indeed...

**Pull request merged**